test/hotspot/jtreg/serviceability/jvmti/ObjectMonitorUsage/libObjectMonitorUsage.cpp

```diff
@@ -0,0 +1,249 @@
+
+  return JNI_OK;
+}
+
+JNIEXPORT jint JNICALL
+Agent_OnLoad(JavaVM *jvm, char *options, void *reserved) {
+  return Agent_Initialize(jvm, options, reserved);
+}
+
+JNIEXPORT jint JNICALL
+Agent_OnAttach(JavaVM *jvm, char *options, void *reserved) {
+  return Agent_Initialize(jvm, options, reserved);
+}
+
+static void print_monitor_info(JNIEnv *jni, jvmtiMonitorUsage &inf) {
+  jvmtiError err;
+  jvmtiThreadInfo tinf;
+
+  LOG(">>> [%d]\n", check_idx);
+  if (inf.owner == nullptr) {
+    LOG(">>>          owner:               none (0x0)\n");
+  } else {
+    err = jvmti->GetThreadInfo(inf.owner, &tinf);
+    check_jvmti_status(jni, err, "error in JVMTI GetThreadInfo");
+    LOG(">>>          owner:               %s (0x%p)\n",
+        tinf.name, inf.owner);
+    deallocate(jvmti, jni, tinf.name);
+  }
+  LOG(">>>          entry_count:         %d\n", inf.entry_count);
+  LOG(">>>          waiter_count:        %d\n", inf.waiter_count);
+  LOG(">>>          notify_waiter_count: %d\n", inf.notify_waiter_count);
+
+  if (inf.waiter_count > 0) {
+    LOG(">>>  waiters:\n");
+    for (int j = 0; j < inf.waiter_count; j++) {
+      err = jvmti->GetThreadInfo(inf.waiters[j], &tinf);
+      check_jvmti_status(jni, err, "error in JVMTI GetThreadInfo");
+      LOG(">>>                %2d: %s (0x%p)\n",
+          j, tinf.name, inf.waiters[j]);
+      deallocate(jvmti, jni, tinf.name);
+    }
+  }
+  if (inf.notify_waiter_count > 0) {
+    LOG(">>>  notify_waiters:\n");
+    for (int j = 0; j < inf.notify_waiter_count; j++) {
+      err = jvmti->GetThreadInfo(inf.notify_waiters[j], &tinf);
+      check_jvmti_status(jni, err, "error in JVMTI GetThreadInfo");
+      LOG(">>>                %2d: %s (0x%p)\n",
+          j, tinf.name, inf.notify_waiters[j]);
+      deallocate(jvmti, jni, tinf.name);
+    }
+  }
+}
+
+JNIEXPORT void JNICALL
+Java_ObjectMonitorUsage_check(JNIEnv *jni, jclass cls, jobject obj, jthread owner,
+        jint entryCount, jint waiterCount, jint notifyWaiterCount) {
+  jvmtiError err;
+  jvmtiMonitorUsage inf;
+
+  check_idx++;
+
+  err = jvmti->GetObjectMonitorUsage(obj, &inf);
+  check_jvmti_status(jni, err, "error in JVMTI GetObjectMonitorUsage");
+
+  print_monitor_info(jni, inf);
+
+  if (!jni->IsSameObject(owner, inf.owner)) {
+    LOG("(%d) unexpected owner: 0x%p\n", check_idx, inf.owner);
+    result = STATUS_FAILED;
+  }
+  if (inf.entry_count != entryCount) {
+    LOG("(%d) entry_count expected: %d, actually: %d\n",
+        check_idx, entryCount, inf.entry_count);
+    result = STATUS_FAILED;
+  }
+  if (inf.waiter_count != waiterCount) {
+    LOG("(%d) waiter_count expected: %d, actually: %d\n",
+        check_idx, waiterCount, inf.waiter_count);
+    result = STATUS_FAILED;
+  }
+  if (inf.notify_waiter_count != notifyWaiterCount) {
+    LOG("(%d) notify_waiter_count expected: %d, actually: %d\n",
+        check_idx, notifyWaiterCount, inf.notify_waiter_count);
+    result = STATUS_FAILED;
```

Comment:
Please consider prefixing these LOG output lines with "FAILED:" so that it
is easier to identify the log output where failure info is shown.

Reply:
Good suggestion. Fixed now.